apps/meteor/server/models/raw/LivechatRooms.ts

```diff
@@ -2761,19 +2763,47 @@ export class LivechatRoomsRaw extends BaseRaw<IOmnichannelRoom> implements ILive
 		throw new Error('Method not implemented.');
 	}
 
-	setContactIdByVisitorAssociation(contactId: string, visitor: ILivechatContactVisitorAssociation): Promise<UpdateResult | Document> {
+	setContactByVisitorAssociation(
+		association: ILivechatContactVisitorAssociation,
+		contact: Pick<AtLeast<ILivechatContact, '_id'>, '_id' | 'name'>,
+	): Promise<UpdateResult | Document> {
 		return this.updateMany(
 			{
 				't': 'l',
-				'v._id': visitor.visitorId,
-				'source.type': visitor.source.type,
-				...(visitor.source.id ? { 'source.id': visitor.source.id } : {}),
+				'v._id': association.visitorId,
+				'source.type': association.source.type,
+				...(association.source.id ? { 'source.id': association.source.id } : {}),
+			},
+			{
+				$set: {
+					'v.contactId': contact._id,
+					...(contact.name ? { fname: contact.name } : {}),
+				},
 			},
-			{ $set: { 'v.contactId': contactId } },
 		);
 	}
 
-	replaceContactId(_oldContactId: string, _newContactId: string): Promise<UpdateResult | Document> {
-		throw new Error('Method not implemented.');
+	updateContactDataByContactId(
+		oldContactId: ILivechatContact['_id'],
+		contact: Partial<Pick<ILivechatContact, '_id' | 'name'>>,
+	): Promise<UpdateResult | Document> {
+		const update = {
+			...(contact._id ? { 'v.contactId': contact._id } : {}),
+			...(contact.name ? { fname: contact.name } : {}),
+		};
+
+		if (!Object.keys(update).length) {
+			throw new Error('error-invalid-operation');
+		}
```

Comment:
It'd be nice if we had some sort of `AtLeastOne` type in `core-typings` so that this kind of error could be avoided
But also not a big deal, we can keep it like this if we need to move fast